### DISCUSSION TITLE
"Automatic Save" for the actual .mm file? (not just the .backups copies)

### DISCUSSION THREAD

**Discussion by hi2u** [freeplane/freeplane#1365] [Help]
* **"Automatic Save"** in Freeplane seems to be a bit different to many other programs...
  * It seems that these settings only relate to the redundant backup copies under the `.backup` dir
  * NOT the actual main `.mm` file that I'm working on, which always remains unsaved until I manually hit `ctrl+s`
* **Question:** Is there a way to get autosave for the main `.mm` file itself?
  * i.e. It would basically give me the same result as immediately hitting `ctrl+s` every time I make any change to the file.  But without me having to hit `ctrl+s`
* I'd still like it to keep creating the `.backup` copies too if possible.

**Comment by euu2021** [comment]
Take a look at the "Multithread" script that I mentioned at https://github.com/freeplane/freeplane/discussions/544#discussion-4286331

For more control over backups, see this discussion and the excellent program created by macmarum: https://github.com/freeplane/freeplane/discussions/1290

**Comment by walmar** [comment]
I would like to have backups without automatically saving the original mind map file as was the case before v1.12.7 (#1500).

**Reply by dpolivaev** [reply]
I see the need but I don't know how to name the option. Can you please suggest it?

**Reply by walmar** [reply]
What I noticed about the latest autosave behavior: In the Preferences window and [documentation](https://docs.freeplane.org/user-documentation/working-w-mindmap-files.html#autosave), the differences between the following file types seem to be a bit unclear to users.
* The ``.mm`` target file and
* the two types of separate copies of the target file:
   * ``.autosave`` live snapshots in the given time interval and
   * ``.bak`` backups at program exit.

Something similar to the following option could be added to the current state:
* Setting: Automatically save target file
   * Notes: If checked, the target file will be overwritten in the given time interval.
      * or: If checked, changes are written to the target file in the given time interval.

Instead of the term *target file* it could also be called *mind map file*.

The setting ``Delete automatic saves at exit`` could be changed to ``Delete autosave files at exit``.

Another feature related to saving maps that I already have on my request list is Notepad++'s ability to persist changes to file contents between program sessions without prompting to save the files. I may open it in the future.

**Reply by dpolivaev** [reply]
I added Auto-save option "Overwrite mind map files automatically" disabled by default in preview 1.12.9_15

**Reply by dpolivaev** [reply]
I also changed some related property descriptions

**Reply by walmar** [reply]
I'm already using the preview. So far so good.
